.github/workflows/test-application.yaml

```diff
@@ -28,3 +33,121 @@ jobs:
+
+    test:
+        name: "Run tests with php ${{ matrix.php-version }} (${{ matrix.phpcr-transport }})"
+        runs-on: ubuntu-latest
+
+        env:
+            JACKRABBIT_VERSION: 2.12.0
+            SYMFONY_DEPRECATIONS_HELPER: weak
+            SYMFONY__DATABASE__HOST: 127.0.0.1
+
+        strategy:
+            fail-fast: false
+            matrix:
+                php-version:
+                    - '5.5'
+                    - '7.0'
+                database: [ mysql ]
+                phpcr-transport: [ doctrinedbal ]
+                composer-flags: [ '' ]
+                test-flags: [ '' ]
+                php-extensions: [ 'ctype, iconv, mysql, gd' ]
+                tools: [ 'composer:v1' ]
+                env:
+                    - {
+                          SYMFONY__PHPCR__TRANSPORT: doctrinedbal,
+                          SYMFONY__DATABASE__DRIVER: pdo_mysql,
+                          SYMFONY__DATABASE__PORT: 3306,
+                    }
+                include:
+                    - php-version: '7.4'
+                      database: postgres
+                      phpcr-transport: jackrabbit
+                      composer-flags: '--prefer-dist --no-interaction'
+                      test-flags: '--jackrabbit-restart' # restart jackrabbit after each suite see: https://github.com/sulu-io/sulu/issues/2137
+                      php-extensions: 'ctype, iconv, mysql, imagick'
+                      tools: 'composer:v2'
+                      env:
+                          SYMFONY__PHPCR__TRANSPORT: jackrabbit
+                          SYMFONY__DATABASE__DRIVER: pdo_pgsql
+                          SYMFONY__DATABASE__PORT: 5432
+                          SYMFONY__DATABASE__USER: postgres
+                          SYMFONY__DATABASE__PASSWORD: postgres
+
+        services:
+            mysql:
+                image: mysql:5.7
+                env:
+                    MYSQL_ALLOW_EMPTY_PASSWORD: yes
+                ports:
+                    - 3306:3306
+                options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=5
+
+            postgres:
+                image: postgres
+                env:
+                    POSTGRES_USER: postgres
+                    POSTGRES_PASSWORD: postgres
+                ports:
+                    - 5432:5432
+                options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+
+        steps:
+            - name: Checkout project
+              uses: actions/checkout@v2
+
+            - name: Install and configure PHP
+              uses: shivammathur/setup-php@v2
+              with:
+                  php-version: ${{ matrix.php-version }}
+                  extensions: ${{ matrix.php-extensions }}
+                  tools: ${{ matrix.tools }}
+                  coverage: none
+
+            - name: Cache jackrabbit runnable
```

Comment:
maybe we should use our jackrabbit image as service?

Reply:
Jackrabbit can't be used as a service, because the `bin/runtests` script needs to be able to restart jackrabbit after each test suite, which is not possible with the current `bin/runtests` script, if jackrabbit is a docker container. In sulu 2.1, it shouldn't be needed anymore to restart jackrabbit, therefore we can then use it as a service

Reply:
I have already mentioned that in our slack channel, if we add much more tests which talk to jackrabbit, that might be necessary again at some point, because it needs to be restarted when jackrabbit has too much load (don't ask me why that is). So it might also be necessary for 2.1 and later at some point, therefore I would stick with the restartable standalone jar in that case.

Reply:
I'm fine with that as well

Reply:
i think it would be nice to use the image. first because it would be consistent to the other services, secondly because it would test the image 🙂 

has anybody tested if these errors till occur? as far as i see, the version of jackrabbit that was previously used on travis is at least 4 years old (https://archive.apache.org/dist/jackrabbit/2.12.0/) maybe this was an issue that was fixed? also, if the issue still persists, i suspect that it is rather a problem with our tests than with jackrabbit? 🙈

Reply:
Haven't tested it, and i don't know if it also happens with a later version.

Reply:
I would just use jackrabbit as service starting from `release/2.1`, because no matter if we use it as service or as standalone package, the same problems will occurr, because the current `bin/runtests` script doesn't handle that yet. So if we ever encounter such an error, we have to modify both, the `bin/runtests` and the `test-application.yaml`